docs/topics/development/setup_and_configuration.md

```diff
@@ -0,0 +1,316 @@
+    - Here’s a high-level overview of what `make up` does:
+
+      ```sh
+      up: setup docker_mysqld_volume_create docker_extract_deps docker_compose_up
+      ```
+
+      - **setup**: Creates configuration files such as `.env`.
+      - **docker_mysqld_volume_create**: Ensures the MySQL volume is created.
+      - **docker_extract_deps**: Installs dependencies inside the Docker container.
+      - **docker_compose_up**: Starts the Docker containers defined in [docker-compose.yml][docker-compose].
+
+5. **Accessing the Development Environment**:
+    - Add the following entry to your `/etc/hosts` file to access **addons-server** via a local domain:
+
+      ```sh
+      127.0.0.1 olympia.test
+      ```
+
+    - The web application should now be accessible at `http://olympia.test`.
+    - You can access the web container for debugging and development:
+
+      ```sh
+      make shell
+      ```
+
+    - To access the Django shell within the container:
+
+      ```sh
+      make djshell
+      ```
+
+### Configuring your environment
+
+#### Build vs Pull
+
+The default behaviour of running addons-server is to build the image locally with a build cache to make subsequent builds
+extremely fast. You can configure your environment to run a pulled image. This can be useful if you have limited CPU
+or if you want to run a very specific version of addons-server, say for testing a Pull request or debugging the version
+currently in a deployed environment.
+
+When you run `make setup` (or make up) the generated .env file determines whether the image should be built or pulled.
+
+Running `make up` without any arguments will produce something like this. Let's look at what these values actually mean.
+
+```bash
+DOCKER_TAG=mozilla/addons-server:local
+DOCKER_TARGET=development
+DOCKER_PULL_POLICY=build
+```
+
+`DOCKER_TAG` specifies the precise tag of the image.
+`DOCKER_TARGET` specifies which target of the image to run. Valid options are `development` or `production`.
+
+> Running addons-server with `DOCKER_TARGET=production` will exclude development dependencies and more closely mirror
+deployed environments, however some dev related functionality may stop working due to missing dependencies.
+
+`DOCKER_PULL_POLICY` determines the build/pull behaviour of docker compose ([learn more](https://docs.docker.com/compose/compose-file/05-services/#pull_policy))
+
+This value is normally determined internally by the DOCKER_TAG to ensure the local image is built and remote images are
+pulled.
+
+Each of these environment variables are interpolated at runtime
+and subsitute areas in the [docker-compose.yml][docker-compose] file
+
+#### Version vs Digest
+
+The default behavior is to build the docker image locally, but if you want to run addons-server with a remote image
+you can specify a docker image version to pull with:
+
+```bash
+make up DOCKER_VERSION=<version>
```

Comment:
can you include a valid example and explain/link where you'd get the list of possible docker image versions from?

Reply:
Done